### ISSUE TITLE
 Visits from optional consent file should hold the same Visits as the mandatory consent.

### ISSUE THREAD

**Issue by Mastaleru** [PharmaLedger-IMI/eco-iot-pmed-workspace#461]
This is a business demand. However let's add a validation on the optional consent, that the visits&procedures  header(visits, date/time windows) are the same with mandatory consents visits

**Comment by Desouler** [comment]
Committed

**Comment by Mastaleru** [comment]
when versioning an optional consent with a valid file
![image](https://user-images.githubusercontent.com/1469129/190693062-d027285f-3f3a-4bd1-807e-c8643135329b.png)

**Comment by Desouler** [comment]
Committed

**Comment by Mastaleru** [comment]
@Desouler I am not able to update the visits from the mandatory consent with a new visit file that holds different visits.
The restriction should apply only to optional consents

**Comment by Desouler** [comment]
Committed

**Comment by Mastaleru** [comment]
Reopened with the following issue for visits&procedures consent:

1. I created the mandatory consent with a visit file VFM1.
2. I updated the mandatory consent with another visit file VFM2.
3. I want to create an optional consent with visit file VFO2 that matches  VFM2 visits => it does not let me to do so (visit file validation constraints )
4. Instead it let me to create the optional consent with VFO1 that matches the obsolete version VFM1 visits,.
